libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentType.java

```diff
@@ -144,10 +137,134 @@ public XContent xContent() {
+                    Map.of(COMPATIBLE_WITH_PARAMETER_NAME, VERSION_PATTERN)),
+                new HeaderValue(VENDOR_APPLICATION_PREFIX + "x-ndjson",
+                    Map.of(COMPATIBLE_WITH_PARAMETER_NAME, VERSION_PATTERN)));
+        }
+
+        @Override
+        public XContentType canonical() {
+            return JSON;
+        }
+    },
+    /**
+     * Versioned jackson based smile binary format. Fast and compact binary format.
+     */
+    VND_SMILE(5) {
+        @Override
+        public String mediaTypeWithoutParameters() {
+            return VENDOR_APPLICATION_PREFIX + "smile";
+        }
+
+        @Override
+        public String queryParameter() {
+            return "vnd_smile";
+        }
+
+        @Override
+        public XContent xContent() {
+            return SmileXContent.smileXContent;
+        }
+
+        @Override
+        public Set<HeaderValue> headerValues() {
+            return Set.of(
+                new HeaderValue(VENDOR_APPLICATION_PREFIX + "smile",
+                    Map.of(COMPATIBLE_WITH_PARAMETER_NAME, VERSION_PATTERN)));
+        }
+
+        @Override
+        public XContentType canonical() {
+            return SMILE;
+        }
+    },
+    /**
+     * A Versioned YAML based content type.
+     */
+    VND_YAML(6) {
+        @Override
+        public String mediaTypeWithoutParameters() {
+            return VENDOR_APPLICATION_PREFIX + "yaml";
+        }
+
+        @Override
+        public String queryParameter() {
+            return "vnd_yaml";
+        }
+
+        @Override
+        public XContent xContent() {
+            return YamlXContent.yamlXContent;
+        }
+
+        @Override
+        public Set<HeaderValue> headerValues() {
+            return Set.of(
+                new HeaderValue(VENDOR_APPLICATION_PREFIX + "yaml",
+                    Map.of(COMPATIBLE_WITH_PARAMETER_NAME, VERSION_PATTERN)));
+        }
+
+        @Override
+        public XContentType canonical() {
+            return YAML;
+        }
+    },
+    /**
+     * A Versioned CBOR based content type.
+     */
+    VND_CBOR(7) {
+        @Override
+        public String mediaTypeWithoutParameters() {
+            return VENDOR_APPLICATION_PREFIX + "cbor";
+        }
+
+        @Override
+        public String queryParameter() {
+            return "vnd_cbor";
+        }
+
+        @Override
+        public XContent xContent() {
+            return CborXContent.cborXContent;
+        }
+
+        @Override
+        public Set<HeaderValue> headerValues() {
+            return Set.of(
                 new HeaderValue(VENDOR_APPLICATION_PREFIX + "cbor",
                     Map.of(COMPATIBLE_WITH_PARAMETER_NAME, VERSION_PATTERN)));
         }
+
+        @Override
+        public XContentType canonical() {
+            return CBOR;
+        }
     };
 
     public static final MediaTypeRegistry<XContentType> MEDIA_TYPE_REGISTRY = new MediaTypeRegistry<XContentType>()
```

Comment:
I think the MediaTypeRegistry class level java doc could use a minor update with this change.